pennylane/devices/tests/test_compare_default_qubit.py

```diff
@@ -68,6 +68,40 @@ def circuit(theta, phi):
         assert np.allclose(qnode(theta, phi), qnode_def(theta, phi), atol=tol(dev.shots))
         assert np.allclose(grad(theta, phi), grad_def(theta, phi), atol=tol(dev.shots))
 
+    def test_projector_expectation(self, device, tol):
+        """Test that arbitrary multi-mode Projector expectation values are correct"""
+        n_wires = 2
+        dev = device(n_wires)
+        dev_def = qml.device("default.qubit", wires=n_wires)
+
+        if dev.shots is not None:
+            pytest.skip("Device is in non-analytical mode.")
+
+        if "Projector" not in dev.observables:
+            pytest.skip("Device does not support the Projector observable.")
+
+        if dev.name == dev_def.name:
+            pytest.skip("Device is default.qubit.")
+
+        theta = 0.432
+        phi = 0.123
+
+        def circuit(theta, phi, basis_state):
+            qml.RX(theta, wires=[0])
+            qml.RX(phi, wires=[1])
+            qml.CNOT(wires=[0, 1])
+            return qml.expval(qml.Projector(basis_state, wires=[0, 1]))
+
+        qnode_def = qml.QNode(circuit, dev_def)
+        qnode = qml.QNode(circuit, dev)
+
+        grad_def = qml.grad(qnode_def, argnum=[0, 1])
+        grad = qml.grad(qnode, argnum=[0, 1])
+
+        for s in [[0, 0], [0, 1], [1, 0], [1, 1]]:
+            assert np.allclose(qnode(theta, phi, s), qnode_def(theta, phi, s), atol=tol(dev.shots))
+            assert np.allclose(grad(theta, phi, s), grad_def(theta, phi, s), atol=tol(dev.shots))
```

Comment:
Nice combination of basis state tests!

Perhaps it would be best to avoid the `for` loop, and instead parametrize this test using the pytest decorator:

```python
@pytest.mark.parametrize("s", [[0, 0], [0, 1], [1, 0], [1, 1]])
def test_projector_expectation(self, device, s, tol):
```

This has two advantages:

1. PyTest would treat it as 4 independent test cases. Unlike the for loop approach, where one failure will cause the test to exist before all 4 cases are tested.

2. Upon a failure, PyTest will display in the console output the particular value of `s` that caused the failure.

Reply:
That totally makes sense! 

This makes me wonder if I should do the same for the other tests in ```test_measurements.py```. 
There are basically four different circuits, so I could define the values of (rotation angles, expected evals, mean and var) at the top, and then use the expected evals, mean and var whenever needed. But this way, only part of the list would be used, and they would be numbers instead of comprehensive formula. So maybe it's better to keep it this way for now.